### PR TITLE
added auth with api key

### PR DESCRIPTION
**Pull request by nabeelamirch**
added auth with api key

resolves: GCI-1450

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthenticationInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+public class ApiKeyAuthenticationInterceptor extends HandlerInterceptorAdapter {
+
+    private static final Logger LOGGER = getLogger();
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+
+        final String authorisedUser = AuthorisationUtil.getAuthorisedIdentity(request);
+        if (authorisedUser == null) {
+            LOGGER.debugRequest(request, "no authorised identity", null);
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthenticationInterceptor.java:22]: Should these `debugRequest()` calls be `infoRequest()` calls?

File: src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import static uk.gov.companieshouse.chd.order.api.logging.LoggingUtils.getLogger;
+
+public class ApiKeyAuthorisationInterceptor extends HandlerInterceptorAdapter {
+
+    private static final Logger LOGGER = getLogger();
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+
+        final String identityType = AuthorisationUtil.getAuthorisedIdentityType(request);
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java:21]: Do we need to call `AuthorisationUtil.hasInternalUserRole()` somewhere too to ensure the key provided represents a internal client with sufficient privileges?

File: src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthenticationInterceptor.java (hunk 1 above)
Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthenticationInterceptor.java:22]: Yes I will change this

File: src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java (hunk 2 above)
Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java:21]: I don't believe so. We are CHS_API_KEY for the authentication. If we were using CHS_INTERNAL_API_KEY then we would have to check for that internal role.

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java:21]: OK, great! 👍

File: src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        if (!identityType.equals(SecurityConstants.API_KEY_IDENTITY_TYPE)) {
+            LOGGER.debugRequest(request, "invalid identity type [" + identityType + "]", null);
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java:24]: Might as well make these `infoRequest()` too?

Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java:24]: Thanks changed it

**Review by VaughanJackson123 — APPROVED** [review]

**Pull request merged**